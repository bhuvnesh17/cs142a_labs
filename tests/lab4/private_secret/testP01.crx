// test that dereference appears appropriately

array x : int[1];
array y : int[2];

func main() : void {
  let y[0] = 0;
  let x[y[0]] = 5;
}
