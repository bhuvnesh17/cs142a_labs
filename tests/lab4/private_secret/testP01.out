ast.DeclarationList(3,1)
  ast.ArrayDeclaration(3,1)[Symbol(x)]
  ast.ArrayDeclaration(4,1)[Symbol(y)]
  ast.FunctionDefinition(6,1)[Symbol(main), []]
    ast.StatementList(7,3)
      ast.Assignment(7,3)
        ast.Index(7,9)
          ast.AddressOf(7,7)[y]
          ast.LiteralInt(7,9)[0]
        ast.LiteralInt(7,14)[0]
      ast.Assignment(8,3)
        ast.Index(8,9)
          ast.AddressOf(8,7)[x]
          ast.Dereference(8,9)
            ast.Index(8,11)
              ast.AddressOf(8,9)[y]
              ast.LiteralInt(8,11)[0]
        ast.LiteralInt(8,17)[5]

